In [46]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.ticker as ticker
import ipywidgets as widget
# import missingno as msno
import seaborn as sns
import plotly.express as px
import plotly.graph_objs as go
from plotly.subplots import make_subplots
from ipywidgets import interact
plt.rcParams['font.family'] = 'Tahoma'
# from termcolor import colored

pd.options.display.max_columns = None
print(f'pandas version = {pd.__version__}')
print(pd.Timestamp.now())

%matplotlib inline
%config InlineBackend.figure_format = 'retina'

pandas version = 1.5.2
2023-04-15 00:04:27.446406


In [47]:
df = pd.read_csv('condo.csv')

In [48]:
df.columns = ['name','post-by','constructed-yr','new','price','space','room','range-from-bts']
pd.set_option('max_colwidth', None)
df.head()

name  \
0  Happy Condo Ratchada 18, กรุงเทพ- ​ซอยรัชดา 18 ถนนประชาราษฎร์บำเพ็ญ แขวงสามเสนนอก เขตห้วยขวาง กรุงเทพมหานคร, สามเสนนอก, ห้วยขวาง, กรุงเทพ   
1  Aspire Rama 9 : แอสปาย พระราม 9, กรุงเทพซอยพระราม 9 แยก 2 ถนนพระราม 9 แขวงห้วยขวาง เขตห้วยขวาง กรุงเทพมหานคร, ห้วยขวาง, ห้วยขวาง, กรุงเทพ   
2                                            Belle Grand Rama 9 : เบลล์ แกรนด์ พระราม 9, กรุงเทพ131 ถนนพระราม 9, ห้วยขวาง, ห้วยขวาง, กรุงเทพ   
3      Royal Nine Residence Condominium ถนนพระราม 9 แขวงบางกะปิ เขตห้วยขวาง กรุงเทพมหานคร 10310, กรุงเทพพระราม 9, บางกะปิ, ห้วยขวาง, กรุงเทพ   
4                                                           ศุภาลัย ปาร์ค เอกมัย-ทองหล่อ, กรุงเทพเพชรบุรีตัดใหม่, บางกะปิ, ห้วยขวาง, กรุงเทพ   

                                      post-by  \
0                ลงประกาศโดย คนึง นพพรพิทักษ์   
1                ลงประกาศโดย Jeerapon Pakutsa   
2                ลงประกาศโดย Jeerapon Pakutsa   
3             ลงประกาศโดย วีรเมศวร์ ไพศาลพงษ์   
4  ลงประกาศโดย Duangporn Leelalumlert (Tracy)   

                       constructed-yr  new      price      space  \
0  <span>ปีที่สร้างเสร็จ: 2015</span>  NaN  1,450,000   29 ตร.ม.   
1  <span>ปีที่สร้างเสร็จ: 2013</span>  NaN  5,550,000   66 ตร.ม.   
2  <span>ปีที่สร้างเสร็จ: 2015</span>  NaN  9,700,000   95 ตร.ม.   
3                                 NaN  NaN  4,690,000  110 ตร.ม.   
4                                 NaN  NaN  6,200,000   71 ตร.ม.   

                                                                                                                                                                                                                 room  \
0  <li class="listing-rooms pull-left"><span class="bed" title="1 ห้องนอน">1 <i class="pg-icon pgicon-bedroom"></i></span> <span class="bath" title="1 ห้องน้ำ">1 <i class="pg-icon pgicon-bathroom"></i></span></li>   
1  <li class="listing-rooms pull-left"><span class="bed" title="2 ห้องนอน">2 <i class="pg-icon pgicon-bedroom"></i></span> <span class="bath" title="2 ห้องน้ำ">2 <i class="pg-icon pgicon-bathroom"></i></span></li>   
2  <li class="listing-rooms pull-left"><span class="bed" title="2 ห้องนอน">2 <i class="pg-icon pgicon-bedroom"></i></span> <span class="bath" title="2 ห้องน้ำ">2 <i class="pg-icon pgicon-bathroom"></i></span></li>   
3  <li class="listing-rooms pull-left"><span class="bed" title="3 ห้องนอน">3 <i class="pg-icon pgicon-bedroom"></i></span> <span class="bath" title="2 ห้องน้ำ">2 <i class="pg-icon pgicon-bathroom"></i></span></li>   
4  <li class="listing-rooms pull-left"><span class="bed" title="2 ห้องนอน">2 <i class="pg-icon pgicon-bedroom"></i></span> <span class="bath" title="1 ห้องน้ำ">1 <i class="pg-icon pgicon-bathroom"></i></span></li>   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 range-from-bts  
0                                                                                                             

In [49]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3001 entries, 0 to 3000
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   name            3001 non-null   object
 1   post-by         3001 non-null   object
 2   constructed-yr  2701 non-null   object
 3   new             66 non-null     object
 4   price           2999 non-null   object
 5   space           3001 non-null   object
 6   room            2990 non-null   object
 7   range-from-bts  3001 non-null   object
dtypes: object(8)
memory usage: 187.7+ KB


In [50]:
# Extract Number
df['post-by'] = df['post-by'].str.replace("ลงประกาศโดย","").str.strip()
df['constructed-yr'] = df['constructed-yr'].str.extract(r"(\d+)")
df['new'] = df['new'].str.extract(r"(\d+)")
df['price'] = pd.to_numeric(df['price'].str.replace(",",""))
df['space'] = df['space'].str.extract(r'(\d+)')
df['space'] = pd.to_numeric(df['space'])


# Replace same objective column 
# Column constructed-yr and new have similar value, column new have data only 2023 yr
df['constructed-yr'] = df['constructed-yr'].fillna(df['new'])

# Categorize room column
    # Extract wanted part
df['room'] = df['room'].str.findall(r'title="(.+?)">')
    # Categorize function
def split_data(row):
    if isinstance(row, list):
        if len(row) == 2:
            return row[0].split()[0], row[1].split()[0], None
        else:
            if 'สตูดิโอ' in row[0]:
                return None, None, row[0].split()[0]
            else:
                return row[0].split()[0], None, None
    else:
        return None, None, None

    # Apply function
df[['bedroom', 'bathroom', 'studio']] = df['room'].apply(lambda x: pd.Series(split_data(x)))
df['studio'] = df['studio'].str.replace('สตูดิโอ','1').astype('Int64')
# Replace Empty data that not considered as NaN
df['range-from-bts'] = df['range-from-bts'].replace('[]',np.nan)
df['range-from-bts'] = df['range-from-bts'].str.extract(r'\d+ นาที \((\d+) ม\.\) ถึง .+')

# Convert type to int
for i in ['bedroom', 'bathroom', 'range-from-bts', 'constructed-yr']:
    df[i] = df[i].astype('Int64')

# Split Condo name
df[[0,1,2,3,4,5]] = df.name.str.split(',',expand=True)
for i in range(0,6):
    df[i] = df[i].str.strip()
not_null_rows = df[[5]].notnull().any(axis=1)
df.dropna(subset=[0,1,2,3,4],inplace=True)
df.drop(index=(df[not_null_rows]).index,inplace=True)
df.drop([5],axis=1,inplace=True)
df.columns

/var/folders/99/kh3w07wx27nc2hb4tswk10xc0000gn/T/ipykernel_3948/193707567.py:47: UserWarning:

Boolean Series key will be reindexed to match DataFrame index.



Index([          'name',        'post-by', 'constructed-yr',            'new',
                'price',          'space',           'room', 'range-from-bts',
              'bedroom',       'bathroom',         'studio',                0,
                      1,                2,                3,                4],
      dtype='object')

In [51]:
df = df[[0,1,2,3,4,'price','space','bedroom','bathroom','studio','constructed-yr','range-from-bts','post-by']]
df.head(20)

0  \
0                                                                    Happy Condo Ratchada 18   
1                                                            Aspire Rama 9 : แอสปาย พระราม 9   
2                                                 Belle Grand Rama 9 : เบลล์ แกรนด์ พระราม 9   
3   Royal Nine Residence Condominium ถนนพระราม 9 แขวงบางกะปิ เขตห้วยขวาง กรุงเทพมหานคร 10310   
4                                                               ศุภาลัย ปาร์ค เอกมัย-ทองหล่อ   
5                                                       TC Green Rama 9 : ทีซี กรีน พระราม 9   
6                                                          Ratchada City 18 : รัชดา ซิตี้ 18   
7                                                       TC Green Rama 9 : ทีซี กรีน พระราม 9   
8                                                            Regent Home 12 : รีเจ้นท์โฮม 12   
10                    The Niche Pride Thonglor-Phetchaburi : เดอะ นิช ไพรด์ ทองหล่อ-เพชรบุรี   
11                                                                Ivy Ratchada : ไอวี่ รัชดา   
12                                                                        City Home Ratchada   
13                                 One 9 Five Asoke - Rama 9 : วัน ไนน์ ไฟว์ อโศก – พระราม 9   
14                                                         Ratchada City 18 : รัชดา ซิตี้ 18   
15                                                         Ratchada City 18 : รัชดา ซิตี้ 18   
16                                                                รัชดา ออร์คิด คอนโดมิเนียม   
17                                               Garden Asoke-Rama 9 : การ์เดน อโศก-พระราม 9   
18                                                            จอห์นนี่ ทาวเวอร์ คอนโดมิเนียม   
19                                                                   Happy Condo Ratchada 18   
21                                                                           Rhythm Ratchada   

                                                                                     1  \
0   กรุงเทพ- ​ซอยรัชดา 18 ถนนประชาราษฎร์บำเพ็ญ แขวงสามเสนนอก เขตห้วยขวาง กรุงเทพมหานคร   
1          กรุงเทพซอยพระราม 9 แยก 2 ถนนพระราม 9 แขวงห้วยขวาง เขตห้วยขวาง กรุงเทพมหานคร   
2                                                               กรุงเทพ131 ถนนพระราม 9   
3                                                                      กรุงเทพพระราม 9   
4                                                               กรุงเทพเพชรบุรีตัดใหม่   
5                                                                   กรุงเทพถนนพระราม 9   
6                                   กรุงเทพถนนรัชดาภิเษก ซอย 18 (ถนนประชาราษฎร์บำเพ็ญ)   
7                                                                   กรุงเทพถนนพระราม 9   
8                                                               กรุงเทพ41 คอนโดมิเนียม   
10                                                          กรุงเทพถนน เพชรบุรีตัดใหม่   
11                                               กรุงเทพซอยรัชดาภิเษก 20 ถนนรัชดาภิเษก   
12                                             กรุงเทพ- ซอยรัชดาภิเษก 10 ถนนรัชดาภิเษก   
13                          กรุงเทพ195 ถนนพระราม 9 ซอย 5 แขวงห้วยขวาง เขตห้วยขวาง กทม.   
14                                  กรุงเทพถนนรัชดาภิเษก ซอย 18 (ถนนประชาราษฎร์บำเพ็ญ)   
15                                  กรุงเทพถนนรัชดาภิเษก ซอย 18 (ถนนประชาราษฎร์บำเพ็ญ)   
16          กรุงเทพ- ถนนสุทธิสารวินิจฉัย แขวงห้วยขวาง เขตห้วยขวาง จังหวัดกรุงเทพมหานคร   
17                                                                   กรุงเทพถนนจตุรทิศ   
18                                                        กรุงเทพประชาอุทิศ ประชาอุทิศ   
19  กรุงเทพ- ​ซอยรัชดา 18 ถนนประชาราษฎร์บำเพ็ญ แขวงสามเสนนอก เขตห้วยขวาง กรุงเทพมหานคร   
21                    กรุงเทพ558 ถนนรัชดาภิเษก แขวงสามเสนนอก เขตห้วยขวาง กรุงเทพมหานคร   

            2         3        4       price  space  bedroom  bathroom  \
0   สามเสนนอก  ห้วยขวาง  กรุงเทพ   1450000.0     29        1         1   
1    ห้วยขวาง  ห้วยขวาง  กรุงเทพ   

In [52]:
df.studio.unique()

<IntegerArray>
[<NA>, 1]
Length: 2, dtype: Int64

In [53]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2880 entries, 0 to 3000
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   0               2880 non-null   object 
 1   1               2880 non-null   object 
 2   2               2880 non-null   object 
 3   3               2880 non-null   object 
 4   4               2880 non-null   object 
 5   price           2878 non-null   float64
 6   space           2880 non-null   int64  
 7   bedroom         2676 non-null   Int64  
 8   bathroom        2670 non-null   Int64  
 9   studio          193 non-null    Int64  
 10  constructed-yr  2709 non-null   Int64  
 11  range-from-bts  1836 non-null   Int64  
 12  post-by         2880 non-null   object 
dtypes: Int64(5), float64(1), int64(1), object(6)
memory usage: 329.1+ KB


## 1. คอนโดในเขตห้วยขวาง

In [54]:
df_dis = df.groupby(df[2]).count()
df_dis

0     1     3     4  price  space  bedroom  bathroom  studio  \
2                                                                            
บางกะปิ     802   802   802   802    802    802      757       756      40   
สามเสนนอก   471   471   471   471    471    471      415       411      54   
ห้วยขวาง   1607  1607  1607  1607   1605   1607     1504      1503      99   

           constructed-yr  range-from-bts  post-by  
2                                                   
บางกะปิ               735             232      802  
สามเสนนอก             438             195      471  
ห้วยขวาง             1536            1409     1607

In [55]:
df_dis = df.groupby(df[2]).count()
fig = px.pie(df_dis, values=df_dis[0], names=df_dis.index, title='เปอร์เซนต์คอนโดในแต่ละแขวงของเขตห้วยขวาง', template='seaborn')
fig.update_traces(textposition='inside', textinfo='value+percent+label')
fig.show()


จากแผนภูมิวงกลมจะเห็นได้ว่า แขวงบางกะปิมีคอนโด 802 แห่ง (27.8%), แขวงสามเสนนอกมีคอนโด 471 แห่ง (16.4%), แขวงห้วยขวางมีคอนโด 1607 แห่ง (55.8%) ปัจจัยที่ทำให้ตำนวนคอนโดแต่ละแขวงมีมากน้อยนั้น มีหลายปัจจัย เช่น ห้างสรรพสินค้า, ขนส่งสาธารณะ, สถานที่ท่องเที่ยวใกล้เคียง เป็นต้น


## 2. ระยะทางระหว่าง station มีผลต่อราคามากน้อยแค่ไหนในเขตห้วยขวาง

In [56]:
def plot_graph(range, df):
    dfrangedrop = df[df['range-from-bts'].notna()]
    dfrangedrop = df[df['constructed-yr'].notna()]

    if range == '0-250':
        df_range = dfrangedrop[dfrangedrop['range-from-bts'] <= 250]
    elif range == '251-500':
        df_range = dfrangedrop[(dfrangedrop['range-from-bts'] > 250) & (dfrangedrop['range-from-bts'] <= 500)]
    elif range == '501-750':
        df_range = dfrangedrop[(dfrangedrop['range-from-bts'] > 500) & (dfrangedrop['range-from-bts'] <= 750)]
    elif range == '751-1000':
        df_range = dfrangedrop[(dfrangedrop['range-from-bts'] > 750) & (dfrangedrop['range-from-bts'] <= 1000)]
    else: 
        df_range = dfrangedrop[dfrangedrop['range-from-bts'] > 1000]

    df_range = df_range.sort_values(by='range-from-bts', ascending=False)
    df_mean = df_range.groupby('constructed-yr')['price'].mean().reset_index()
    return go.Scatter(x=df_mean['constructed-yr'], y=df_mean['price'], mode='lines', name=f"Distance {range}")

fig = go.Figure()

for range_option in ['0-250', '251-500', '501-750', '751-1000', '>1000']:
    fig.add_trace(plot_graph(range_option, df))

fig.update_layout(
    updatemenus=[
        dict(
            active=0,
            buttons=list([
                dict(
                    label='All distances',
                    method='update',
                    args=[{"visible": [True, True, True, True, True]}]
                ),
                dict(
                    label='0-250m',
                    method='update',
                    args=[{"visible": [True, False, False, False, False]}]
                ),
                dict(
                    label='251-500m',
                    method='update',
                    args=[{"visible": [False, True, False, False, False]}]
                ),
                dict(
                    label='501-750m',
                    method='update',
                    args=[{"visible": [False, False, True, False, False]}]
                ),
                dict(
                    label='751-1000m',
                    method='update',
                    args=[{"visible": [False, False, False, True, False]}]
                ),
                dict(
                    label='>1000m',
                    method='update',
                    args=[{"visible": [False, False, False, False, True]}]
                )
            ])
        )
    ],
    xaxis_title='Constructed year',
    yaxis_title='Price',
    title='How much does the distance between the BTS station and the property affect the price in Huai Khwang district?'
)

# Show the figure
fig.show()


ผลสำรวจการเลือกซื้อคอนโดของ Ananda Development พบว่าคนไทยมากกว่าครึ่งใช้ระยะห่างระหว่างคอนโดกับรถไฟฟ้าเป็นปัจจัยในการตัดสินใจในการเลือกซื้อ เป็นเหตุผลที่ทำให้ราคาคอนโดที่มีระยะห่างระหว่างรถไฟฟ้าน้อย มักมีราคาสูงกว่าคอนโดในระยะห่างรถไฟฟ้าที่มากกว่า ซึ่งสาเหตุหลัก ๆ คือ
1. คอนโดติดรถไฟฟ้าง่ายต่อการลงทุนหรือปล่อยเช่า
2. ประหยัดเวลาการเดินทาง
3. เดินทางสะดวกปลอดภัย
4. Fix ค่าใช้จ่ายได้ทุกเดือน

(หาข้อมูลเพิ่มจาก -> ขึ้นอยู่กับว่าจะเอาไป story telling ยังไง
1. ข้อดีของการซื้อคอนโดติดรถไฟฟ้า
2. ข้อด้อยของการซื้อคอนโดติดรถไฟฟ้า
3. behavior ของคนอยู่คอนโด
4. demand ของคอนโดที่เพิ่มขึ้น/ลดลงในแต่ละปี)

## 3.  ราคาของคอนโดแต่ละห้องในแต่ละแขวงในแต่ละปี
note: แขวงเดียวกันเอาสีเดียวกันจมั๊ยจะได้ไม่สับสน

In [57]:
dfyear = df.copy()
dfyear['bedroom'] = dfyear['bedroom'].astype(str).replace('<NA>', 'studio')
df_avg_price = dfyear.groupby([2, 'constructed-yr', 'bedroom'])['price'].mean().reset_index()
dfavg.rename(columns={2: "sub-district",})
df_avg_price.head(10)

2  constructed-yr bedroom         price
0  บางกะปิ            1998       1  1.490000e+06
1  บางกะปิ            2009       1  4.756500e+06
2  บางกะปิ            2009       2  6.282500e+06
3  บางกะปิ            2009  studio  3.000000e+06
4  บางกะปิ            2010       1  2.664545e+06
5  บางกะปิ            2010       2  5.900000e+06
6  บางกะปิ            2010  studio  1.540000e+06
7  บางกะปิ            2012       1  3.513180e+06
8  บางกะปิ            2012       2  6.789403e+06
9  บางกะปิ            2013       1  6.760000e+06

In [70]:
dfavg = dfyear.groupby([2, 'constructed-yr', 'bedroom'])['price'].mean().reset_index()
dfavg = dfavg.rename(columns={2: "sub-district",})
dfavg.head(10)
# data = dfavg[(dfavg[2] == 'ห้วยขวาง') & (dfavg['bedroom'] == '1')]
# data.groupby('constructed-yr').mean()
dfavg[dfavg['bedroom'] == '1'].groupby('constructed-yr').mean().values

/var/folders/99/kh3w07wx27nc2hb4tswk10xc0000gn/T/ipykernel_3948/3105445300.py:6: FutureWarning:

The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.



array([[1390000.        ],
       [1490000.        ],
       [2076086.95652174],
       [2279062.5       ],
       [2439706.89655172],
       [3717509.08795846],
       [2778655.0802139 ],
       [3802059.92465753],
       [2642369.18964077],
       [4315719.33218391],
       [3111915.05141505],
       [3824982.35714286],
       [3211087.91861166],
       [3393055.64705882],
       [4019652.54032258],
       [3073898.69907407],
       [3636722.35897436],
       [4186709.41176471],
       [4644368.11831832],
       [5340000.        ],
       [3269750.        ]])

In [72]:
fig = make_subplots(rows=4, cols=2,subplot_titles=[
    '1 Bedroom','2 Bedroom','3 Bedroom','4 Bedroom','5 Bedroom','6 Bedroom','Studio'])
d1 = dfavg[dfavg['bedroom'] == '1'].groupby(['constructed-yr','sub-district']).mean(numeric_only=True).reset_index()
d2 = dfavg[dfavg['bedroom'] == '2'].groupby(['constructed-yr','sub-district']).mean(numeric_only=True).reset_index()
d3 = dfavg[dfavg['bedroom'] == '3'].groupby(['constructed-yr','sub-district']).mean(numeric_only=True).reset_index()
d4 = dfavg[dfavg['bedroom'] == '4'].groupby(['constructed-yr','sub-district']).mean(numeric_only=True).reset_index()
d5 = dfavg[dfavg['bedroom'] == '5'].groupby(['constructed-yr','sub-district']).mean(numeric_only=True).reset_index()
d6 = dfavg[dfavg['bedroom'] == '6'].groupby(['constructed-yr','sub-district']).mean(numeric_only=True).reset_index()
ds = dfavg[dfavg['bedroom'] == 'studio'].groupby(['constructed-yr','sub-district']).mean(numeric_only=True).reset_index()
#1
fig.add_trace(go.Scatter(x=d1[d1['sub-district']=='ห้วยขวาง']['constructed-yr'],
                         y=d1[d1['sub-district']=='ห้วยขวาง']['price'],
                         text='ห้วยขวาง'
                         ),row=1,col=1)
fig.add_trace(go.Scatter(x=d1[d1['sub-district']=='บางกะปิ']['constructed-yr'],
                         y=d1[d1['sub-district']=='บางกะปิ']['price'],
                         text='บางกะปิ'
                         ),row=1,col=1)
fig.add_trace(go.Scatter(x=d1[d1['sub-district']=='สามเสนนอก']['constructed-yr'],
                         y=d1[d1['sub-district']=='สามเสนนอก']['price'],
                         text='สามเสนนอก'
                         ),row=1,col=1)
#2
fig.add_trace(go.Scatter(x=d2[d2['sub-district']=='ห้วยขวาง']['constructed-yr'],
                         y=d2[d2['sub-district']=='ห้วยขวาง']['price'],
                         text='ห้วยขวาง'
                         ),row=1,col=2)
fig.add_trace(go.Scatter(x=d2[d2['sub-district']=='บางกะปิ']['constructed-yr'],
                         y=d2[d2['sub-district']=='บางกะปิ']['price'],
                         text='บางกะปิ'
                         ),row=1,col=2)
fig.add_trace(go.Scatter(x=d2[d2['sub-district']=='สามเสนนอก']['constructed-yr'],
                         y=d2[d2['sub-district']=='สามเสนนอก']['price'],
                         text='สามเสนนอก'
                         ),row=1,col=2)
#3
fig.add_trace(go.Scatter(x=d3[d3['sub-district']=='ห้วยขวาง']['constructed-yr'],
                         y=d3[d3['sub-district']=='ห้วยขวาง']['price'],
                         text='ห้วยขวาง'
                         ),row=2,col=1)
fig.add_trace(go.Scatter(x=d3[d3['sub-district']=='บางกะปิ']['constructed-yr'],
                         y=d3[d3['sub-district']=='บางกะปิ']['price'],
                         text='บางกะปิ'
                         ),row=2,col=1)
fig.add_trace(go.Scatter(x=d3[d3['sub-district']=='สามเสนนอก']['constructed-yr'],
                         y=d3[d3['sub-district']=='สามเสนนอก']['price'],
                         text='สามเสนนอก'
                         ),row=2,col=1)
#4
fig.add_trace(go.Scatter(x=d4[d4['sub-district']=='ห้วยขวาง']['constructed-yr'],
                         y=d4[d4['sub-district']=='ห้วยขวาง']['price'],
                         text='ห้วยขวาง'
                         ),row=2,col=2)
fig.add_trace(go.Scatter(x=d4[d4['sub-district']=='บางกะปิ']['constructed-yr'],
                         y=d4[d4['sub-district']=='บางกะปิ']['price'],
                         text='บางกะปิ'
                         ),row=2,col=2)
fig.add_trace(go.Scatter(x=d4[d4['sub-district']=='สามเสนนอก']['constructed-yr'],
                         y=d4[d4['sub-district']=='สามเสนนอก']['price'],
                         text='สามเสนนอก'
                         ),row=2,col=2)
#5
fig.add_trace(go.Scatter(x=d5[d5['sub-district']=='ห้วยขวาง']['constructed-yr'],
                         y=d5[d5['sub-district']=='ห้วยขวาง']['price'],
                         text='ห้วยขวาง'
                         ),row=3,col=1)
fig.add_trace(go.Scatter(x=d5[d5['sub-district']=='บางกะปิ']['constructed-yr'],
                         y=d5[d5['sub-district']=='บางกะปิ']['price'],
                         text='บางกะปิ'
                         ),row=3,col=1)
fig.add_trace(go.Scatter(x=d5[d5['sub-district']=='สามเสนนอก']['constructed-yr'],
                         y=d5[d5['sub-district']=='สามเสนนอก']['price'],
                         text='สามเสนนอก'
                         ),row=3,col=1)
#6
fig.add_trace(go.Scatter(x=d6[d6['sub-district']=='ห้วยขวาง']['constructed-yr'],
                         y=d6[d6['sub-district']=='ห้วยขวาง']['price'],
                         text='ห้วยขวาง'
                         ),row=3,col=2)
fig.add_trace(go.Scatter(x=d6[d6['sub-district']=='บางกะปิ']['constructed-yr'],
                         y=d6[d6['sub-district']=='บางกะปิ']['price'],
                         text='บางกะปิ'
                         ),row=3,col=2)
fig.add_trace(go.Scatter(x=d5[d5['sub-district']=='สามเสนนอก']['constructed-yr'],
                         y=d5[d5['sub-district']=='สามเสนนอก']['price'],
                         text='สามเสนนอก'
                         ),row=3,col=2)
#Studio
fig.add_trace(go.Scatter(x=ds[ds['sub-district']=='ห้วยขวาง']['constructed-yr'],
                         y=ds[ds['sub-district']=='ห้วยขวาง']['price'],
                         text='ห้วยขวาง'
                         ),row=4,col=1)
fig.add_trace(go.Scatter(x=ds[ds['sub-district']=='บางกะปิ']['constructed-yr'],
                         y=ds[ds['sub-district']=='บางกะปิ']['price'],
                         text='บางกะปิ'
                         ),row=4,col=1)
fig.add_trace(go.Scatter(x=ds[ds['sub-district']=='สามเสนนอก']['constructed-yr'],
                         y=ds[ds['sub-district']=='สามเสนนอก']['price'],
                         text='สามเสนนอก'
                         ),row=4,col=1)

fig.update_layout(
    title='Condo price by Room and sub-district',
    # yaxis=dict(title='Values'),
    height=900,
    width=900,
    showlegend=False
)
fig.show()

เดี๋ยวคิดเพิ่มตอนทำ story telling คิดบ่ออก
- ราคาของคอนโดมีแนวโน้มจะเพิ่มขึ้นโดยเฉพาะห้อง studio เนื่องจาก
- จะเห็นได้ว่าแขวงห้วยขวางมีคอนโดให้เลือกทุกรูปแบบและมีแนวโน้มราคาที่จะเพิ่มขึ้นเรื่อย ๆ ทุกปี สาเหตุจากเป็นแหล่งที่อยู่อาศัยที่เพียบพร้อมไปด้วยสิ่งอำนวยความสะดวกมากมาย เช่น  สถานี mrt โรงพยาบาลหัวเฉียว ทางด่วน ห้างสรรพสินค้าอย่างเซ็นทรัลพระราม 9 และthe street ratchada ส่งผลให้ราคาที่ดินในแต่ละปีเพิ่มสูงขึ้นเรื่อย ๆ จึงทำให้ราคาคอนโดในแขวงห้วยขวางเพิ่มสูงขึ้นตามด้วย
- แขวงสามเสนนอก ติดสถานี mrt แค่ 2 สถานี ต่างจากแขวงห้วยขวาง ภายในแขวงนี้ไม่มีสิ่งอำนวยความสะดวมากมาย อย่างห้างสรรพสินค้า หรือโรงพยาบาลใกล้เคียง บริเวณนี้ส่วนใหญ่เป็นบ้านพักอาศัยจึงไม่ค่อยมีคอนโดบริเวณนี้ ราคาของคอนโดบริเวณนี้จึงไม่สูงมาก
- แขวงบางกะปิ แขวงที่อยู่อาศัยที่มีทั้งคอนโดและบ้านเดี่ยวจำนวนมาก มีสถานี mrt สถานีเพชรบุรีภายในแขวง แต่มีรถไฟฟ้า airport link ซึ่งเป็นรถไฟฟ้าที่ไปสู่ทางเชื่อม รถไฟไทย mrt และ BTS ตัดผ่านตลอดทาง นอกจากสิ่งอำนวยความสะดวกดังกล่าวแล้วยังมีโรงพยาบาลมากมาย เช่น โรงพยาบาลพระราม9 โรงพยาบาลปิยเวท เป็นต้น มีทางขึ้นทางด่วนสะดวกสบาย แต่ย่านนี้มีห้างแค่ show dc ราคาย่านนี้จึงอยู่ในระดับกลาง